python/dataproc_templates/gcs/gcs_to_gcs.py

```diff
@@ -0,0 +1,197 @@
+        
+        parser.add_argument(
+            f'--{constants.GCS_TO_GCS_OUTPUT_MODE}',
+            dest=constants.GCS_TO_GCS_OUTPUT_MODE,
+            required=False,
+            default=constants.OUTPUT_MODE_APPEND,
+            help=(
+                'Output write mode '
+                '(one of: append,overwrite,ignore,errorifexists) '
+                '(Defaults to append)'
+            ),
+            choices=[
+                constants.OUTPUT_MODE_OVERWRITE,
+                constants.OUTPUT_MODE_APPEND,
+                constants.OUTPUT_MODE_IGNORE,
+                constants.OUTPUT_MODE_ERRORIFEXISTS
+            ]
+        )
+        
+        parser.add_argument(
+            f'--{constants.GCS_TO_GCS_OUTPUT_LOCATION}',
+            dest=constants.GCS_TO_GCS_OUTPUT_LOCATION,
+            required=True,
+            help=(
+                'destination GCS location'
+            )
+        )
+        
+        known_args: argparse.Namespace
+        known_args, _ = parser.parse_known_args(args)
+
+        return vars(known_args)
+    
+    def run(self, spark: SparkSession, args: Dict[str, Any]) -> None:
+
+        logger: Logger = self.get_logger(spark=spark)
+        # Arguments
+        gcs_input_location: str = args[constants.GCS_TO_GCS_INPUT_LOCATION]
+        gcs_input_format: str = args[constants.GCS_TO_GCS_INPUT_FORMAT]
+        gcs_temp_view: str = args[constants.GCS_TO_GCS_TEMP_VIEW_NAME]
+        sql_query: str = args[constants.GCS_TO_GCS_SQL_QUERY]
+        output_partition_column: str = args[constants.GCS_TO_GCS_OUTPUT_PARTITION_COLUMN]
+        gcs_output_mode: str = args[constants.GCS_TO_GCS_OUTPUT_MODE]
+        gcs_output_format: str = args[constants.GCS_TO_GCS_OUTPUT_FORMAT]
+        gcs_output_location: str = args[constants.GCS_TO_GCS_OUTPUT_LOCATION]
+        
+        logger.info(
+            "Starting GCS to GCS with tranformations spark job with parameters:\n"
+            f"{pprint.pformat(args)}"
+        )
+        
+        # Read
+        input_data: DataFrame
+        
+        if gcs_input_format == constants.FORMAT_PRQT:
+            input_data = spark.read \
+                .parquet(gcs_input_location)
+        elif gcs_input_format == constants.FORMAT_AVRO:
+            input_data = spark.read \
+                .format(constants.FORMAT_AVRO) \
+                .load(gcs_input_location)
+        elif gcs_input_format == constants.FORMAT_CSV:
+            input_data = spark.read \
+                .format(constants.FORMAT_CSV) \
+                .option(constants.HEADER, True) \
+                .option(constants.INFER_SCHEMA, True) \
+                .load(gcs_input_location)
+        elif gcs_input_format == constants.FORMAT_JSON:
+            input_data = spark.read \
+                .json(gcs_input_location)
+        
+        if ((gcs_temp_view != "") & (sql_query != "")):
+            # Create temp view on source data
+            input_data.createOrReplaceTempView(gcs_temp_view)
```

Comment:
What is the use of registering the table as temp view here?

Reply:
The idea is that if the user wishes to just change the file formats without any transformations, they would not have to provide a query or a temp view name and can easily go ahead with the file format conversion using the same template

Reply:
@snrssc , @shashank-google are any changes required? or are we good to close this?

Reply:
Hi @varunika  , Two options i see here,

1. We can make it as optional property and use a default value if nothing specified
2.  Always use default value as the tempview name is not to be accessible outside of the program and in downstream processing.

Reply:
Discussed in a call, user has to specify same table name in both tempview and the query provided. For now this template is designed only for single GCS file read and write.  @varunika  to add comments for specifying tempview has to match with tablename used in the query.

Reply:
Thanks @varunika . LGTM

Reply:
@snrssc Added new commit to accommodate the requested change. Please review.